test/integration/services/storage/storage-manger.spec.ts

```diff
@@ -0,0 +1,341 @@
+import config from 'config'
+import chai from 'chai'
+import sinonChai from 'sinon-chai'
+import BigNumber from 'bignumber.js'
+import { asciiToHex, hexToAscii } from 'web3-utils'
+
+import { getFeatherClient, TestingApp, ZERO_ADDRESS } from '../../utils'
+import Offer from '../../../../src/services/storage/models/offer.model'
+import BillingPlan from '../../../../src/services/storage/models/billing-plan.model'
+import Rate from '../../../../src/services/rates/rates.model'
+import Agreement from '../../../../src/services/storage/models/agreement.model'
+import StakeModel from '../../../../src/services/storage/models/stake.model'
+import { ServiceAddresses } from '../../../../src/definitions'
+import { WEI } from '../../../../src/services/storage/utils'
+
+chai.use(sinonChai)
+const expect = chai.expect
+
+const generateMsg = (peerId = 'FakePeerId') => {
+  const testPeerIdHex = asciiToHex(peerId, 32).replace('0x', '')
+  const nodeIdFlag = '01'
+  return `0x${nodeIdFlag}${testPeerIdHex}`
+}
+
+function randomStr (length = 32): string {
+  let result = ''
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+  const charactersLength = characters.length
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength))
+  }
+  return result
+}
+
+function generateCID (): string[] {
+  return [asciiToHex(`/ipfs/${randomStr(26)}`)]
+}
+
+describe('Storage service', function () {
+  this.timeout(60000)
+  let app: TestingApp
+
+  before(async () => {
+    // @ts-ignore
+    config.rns.enabled = false
+    app = new TestingApp()
+    await app.initAndStart()
+  })
+  after(async () => {
+    await app.stop()
+    // @ts-ignore
+    config.rns.enabled = true
+  })
+  describe('Storage manager', () => {
+    describe('Offers', () => {
+      afterEach(async () => {
+        await BillingPlan.destroy({ where: {} })
+        await Offer.destroy({ where: {} })
+      })
+      it('should create an offer and plans', async () => {
+        const offerData = {
+          totalCapacity: '1024',
+          periods: [10],
+          prices: [100],
+          msg: generateMsg('test')
```

Comment:
Lets use some real-looking PeerId so we can check that the processing works as expected...

Reply:
Done